src/board.ts

```diff
@@ -3,10 +3,11 @@ import { pos2key, key2pos, opposite, distanceSq, allPos, computeSquareCenter } f
 import { premove, queen, knight } from './premove'
 import * as cg from './types'
 
-export type Callback = (...args: any[]) => void;
-
-export function callUserFunction(f: Callback | undefined, ...args: any[]): void {
-  if (f) setTimeout(() => f(...args), 1);
+export function callUserFunction<T  extends (...args: any) => void>(
+  f: T | undefined,
+  ...args: Parameters<T>
+): void {
+  if (f) setTimeout(() => f(args), 1);
```

Comment:
Wow, really cool features. Is `...args` -> `args` intended here?

Reply:
Thank you
Yes it is intended
https://www.typescriptlang.org/docs/handbook/utility-types.html#parameterstype

Reply:
Ok :+1:

First time I see type annotations changing semantics.

Reply:
Actually ... looking at the generated code, it does not seem right: https://github.com/ornicar/chessground/pull/157/files#diff-930f9993ca95523e8be4573892f7254508010d0e452834b60403d3e6e3b96761R185-R188

Reply:
You are right , I will doubecheck